### PR TITLE
FIrst Feature > Pin screen

### PR DESCRIPTION
**Pull request by katerynaYaryfa**

### CODE REVIEW

File: lib/app.dart [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 class App extends StatelessWidget {
   const App({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return ChangeNotifierProvider<PinNotifier>(
-      create: (context) => PinNotifier(),
+    return MultiProvider(
+      providers: [
+        ChangeNotifierProvider<PinNotifier>(create: (_) => PinNotifier()),
```
Comment (VovaMedvediev) [inline lib/app.dart]: add trailing comma

File: lib/app.dart [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+        ChangeNotifierProvider<SettingsNotifier>(
+            create: (_) => SettingsNotifier())
```
Comment (VovaMedvediev) [inline lib/app.dart]: add trailing comma

File: lib/main.dart [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import 'package:flutter/material.dart';
 import 'package:mood_tracker/app.dart';
 
-void main() {
+void main() async {
```
Comment (VovaMedvediev) [inline lib/main.dart]: please remove async, it is not needed here (you don't have async functions)

File: lib/pin_screen/pin_password_input_field.dart [hunk 4]
```diff
@@ -1,27 +1,75 @@
...
   @override
   State<PasswordInputField> createState() => _PasswordInputFieldState();
 }
 
 class _PasswordInputFieldState extends State<PasswordInputField> {
   @override
   Widget build(BuildContext context) {
+    bool wrongPin = context.watch<PinNotifier>().wrongPin;
+    var color = context.watch<PinNotifier>().color;
+    // var changeColor = context.watch()<PinNotifier>().colorButton();
```
Comment (VovaMedvediev) [inline lib/pin_screen/pin_password_input_field.dart]: please remove comments, it is called "code smell"

File: lib/pin_screen/pin_password_input_field.dart [hunk 5]
```diff
@@ -1,27 +1,75 @@
...
+
+    // bool pin = context.watch<PinNotifier>().pin;
+
+    // var color;
+
+    void colorButton() {
```
Comment (VovaMedvediev) [inline lib/pin_screen/pin_password_input_field.dart]: move this function to class level and add encapsulation (make this function private)

File: lib/pin_screen/pin_password_input_field.dart [hunk 6]
```diff
@@ -1,27 +1,75 @@
...
+        color = Color(0xFF434343);
+      }
+      if (wrongPin = true) {
+        color = Colors.red;
+      } else {
+        Colors.grey.shade400;
+      }
+    }
+
+    final bool1 = true;
```
Comment (VovaMedvediev) [inline lib/pin_screen/pin_password_input_field.dart]: delete dead code

File: lib/pin_screen/pin_password_input_field.dart [hunk 7]
```diff
@@ -1,27 +1,75 @@
...
+    //         ? ''
+    //         : ''
+    //     : 'grey color';
+    // Color color;
+
+    final isUserLogin = false;
+    if (true) {
+    } else {}
+
+    Color pinColor(int num) {
```
Comment (VovaMedvediev) [inline lib/pin_screen/pin_password_input_field.dart]: make private function too

File: lib/pin_screen/pin_password_input_field.dart [hunk 8]
```diff
@@ -1,27 +1,75 @@
...
         Container(
           width: 15,
           height: 15,
           decoration: BoxDecoration(
-            color:
-                widget.pin.length >= 1 ? Colors.blueGrey : Colors.grey.shade400,
-            borderRadius: BorderRadius.circular(100.0),
+            border: Border.all(
+                width: 2.0,
+                color: wrongPin ? Color(0xFFFF7562) : Color(0xFF434343)),
```
Comment (VovaMedvediev) [inline lib/pin_screen/pin_password_input_field.dart]: comma

File: lib/pin_screen/pin_screen.dart [hunk 9]
```diff
@@ -4,28 +4,58 @@ import 'package:mood_tracker/pin_screen/pin_buttons.dart';
...
+    });
+  }
 
   @override
   Widget build(BuildContext context) {
     final pin1 = context.watch<PinNotifier>().pin1;
     final pin2 = context.watch<PinNotifier>().pin2;
     var wrongPin = context.watch<PinNotifier>().wrongPin;
 
+    print('_______pin 1 = ${pin1}');
```
Comment (VovaMedvediev) [inline lib/pin_screen/pin_screen.dart]: remove prints

File: lib/pin_screen/provider/pin_notifier.dart [hunk 10]
```diff
@@ -1,25 +1,44 @@
 import 'package:flutter/material.dart';
+import 'package:mood_tracker/pin_screen/secure_storage.dart';
 
-class PinNotifier with ChangeNotifier {
-  String oldPin = '';
-  Color color1 = Colors.grey.shade400;
-  Color color2 = Colors.grey.shade400;
-  Color color3 = Colors.grey.shade400;
-  Color color4 = Colors.grey.shade400;
+enum PinScreenStates { login, setNewPin, confirmPin, error, success, empty }
```
Comment (VovaMedvediev) [inline lib/pin_screen/provider/pin_notifier.dart]: delete

File: pubspec.yaml [hunk 11]
```diff
@@ -34,7 +34,8 @@ dependencies:
   provider: ^6.0.3
   flutter_secure_storage: ^5.0.2
   flutter_svg: ^1.1.1+1
-
+#  get: ^4.6.5
```
Comment (VovaMedvediev) [inline pubspec.yaml]: remove

**Review by VovaMedvediev — CHANGES_REQUESTED** [review]

File: lib/app.dart (hunk 1 above)
Comment (katerynaYaryfa) [reply lib/app.dart]: added

File: lib/app.dart (hunk 2 above)
Comment (katerynaYaryfa) [reply lib/app.dart]: added

File: lib/main.dart (hunk 3 above)
Comment (katerynaYaryfa) [reply lib/main.dart]: removed

File: lib/pin_screen/pin_password_input_field.dart (hunk 4 above)
Comment (katerynaYaryfa) [reply lib/pin_screen/pin_password_input_field.dart]: removed

File: lib/pin_screen/provider/pin_notifier.dart (hunk 10 above)
Comment (katerynaYaryfa) [reply lib/pin_screen/provider/pin_notifier.dart]: deleted

File: lib/pin_screen/pin_screen.dart (hunk 9 above)
Comment (katerynaYaryfa) [reply lib/pin_screen/pin_screen.dart]: removed

File: lib/pin_screen/pin_password_input_field.dart (hunk 8 above)
Comment (katerynaYaryfa) [reply lib/pin_screen/pin_password_input_field.dart]: added

File: lib/pin_screen/pin_password_input_field.dart (hunk 7 above)
Comment (katerynaYaryfa) [reply lib/pin_screen/pin_password_input_field.dart]: makes

File: pubspec.yaml (hunk 11 above)
Comment (katerynaYaryfa) [reply pubspec.yaml]: removed

**Review by VovaMedvediev — APPROVED** [review]

**Pull request merged**